src/network/networkd-route.c

```diff
@@ -617,6 +618,12 @@ int route_configure(
         if (r < 0)
                 return log_error_errno(r, "Could not append RTA_PREF attribute: %m");
 
+        if (route->lifetime != SEC_INFINITY && link->manager->route_expiration_kernel) {
+                r = sd_netlink_message_append_u32(req, RTA_EXPIRES, route->lifetime);
```

Comment:
hmm, i am pretty sure the `lifetime` field of Route should go, and instead we should calculate the expiry here like this:

`DIV_ROUND_UP(usec_sub(route->expire_time, now(clock_boottime_or_monotonic()), USEC_PER_SEC);`

I.e. calculate the remaining time in a saturated way at the moment where we set this field. That way we don't need the liftime field at all